### PR TITLE
Support Standard Disk Plan for Disk creation

### PR DESCRIPTION
**Pull request by noralife**
This patch enables Standard Disk plan for Disk creation.
Please kindly review this.

### CODE REVIEW

File: lib/fog/sakuracloud/requests/volume/create_disk.rb [hunk 1]
```diff
@@ -30,7 +28,7 @@ def create_disk( name, plan, sourcearchive )
       end # Real
 
       class Mock
-        def create_disk( name, plan, sourcearchive )
+        def create_disk( name, plan, sourcearchive, options = {} )
```
Comment (plribeiro3000) [inline lib/fog/sakuracloud/requests/volume/create_disk.rb]: The `Real` one does not have the same signature as this. Can you fix it?

File: lib/fog/sakuracloud/models/volume/disk.rb [hunk 2]
```diff
@@ -20,8 +20,12 @@ def delete
...
 
         def save
-          requires :name, :plan, :source_archive
-          data = service.create_disk(@attributes[:name], @attributes[:plan], @attributes[:source_archive]).body["Disk"]
+          requires :name, :plan
+          options = {
+            :source_archive=>@attributes[:source_archive],
+            :size_mb=>@attributes[:size_mb]
+          } 
+          data = service.create_disk(@attributes[:name], @attributes[:plan], options).body["Disk"]
```
Comment (plribeiro3000) [inline lib/fog/sakuracloud/models/volume/disk.rb]: You don´t need to use the instance variable `@attributes`. Each attribute has its own getter with the same signature of the attribute so instead of `@attributes[:name]` you can use just `name`.

Comment (plribeiro3000) [inline lib/fog/sakuracloud/models/volume/disk.rb]: Also i would say that you should not work towards parsing of the request in your model. The request should do it for you. So `.body["Disk"]` should be the return of the method `create_disk`.

File: lib/fog/sakuracloud/models/compute/servers.rb [hunk 3]
```diff
@@ -44,10 +44,12 @@ def create_and_attach_volume(server, options)
             Fog::Logger.warning("Create Volume")
             sakuracloud_api_token        = options[:sakuracloud_api_token] || Fog.credentials[:sakuracloud_api_token]
             sakuracloud_api_token_secret = options[:sakuracloud_api_token_secret] || Fog.credentials[:sakuracloud_api_token_secret]
+            size_mb = options[:volume][:size_mb].to_i || 20480
```
Comment (plribeiro3000) [inline lib/fog/sakuracloud/models/compute/servers.rb]: Why `20480`?

Comment (noralife) [inline lib/fog/sakuracloud/models/compute/servers.rb]: I referred official API document.
http://developer.sakura.ad.jp/cloud/api/1.1/disk/

Unfortunately, it is written in Japanese.
Following JSON is the source I referred.

```
{
  "Disk":{
    "ID":"XXXXXXXXXXXXX",
    "Name":"New Disk",
    "Description":"Creating an empty disk",
    "Connection":"virtio",
    "ConnectionOrder":null,
    "ReinstallCount":0,
    "Availability":"available",
    "SizeMB":20480,
    "MigratedMB":null,
    "WaitingJobCount":null,
```

File: lib/fog/sakuracloud/requests/volume/create_disk.rb (hunk 1 above)
Comment (noralife) [inline lib/fog/sakuracloud/requests/volume/create_disk.rb]: Thank you for valuable feedbacks.
It was wrong... It should be
```
      class Mock
        def create_disk( name, plan, options = {} )
```

I will commit later.

File: lib/fog/sakuracloud/models/volume/disk.rb (hunk 2 above)
Comment (noralife) [inline lib/fog/sakuracloud/models/volume/disk.rb]: I will change instance variables to use the getters.
Regarding return of method `create_disk`, a testcase `./tests/sakuracloud/requests/volume/disks_tests.rb`, requires `disks.body.is_a?`
See following codes:

./tests/sakuracloud/requests/volume/disks_tests.rb
```
Shindo.tests('Fog::Volume[:sakuracloud] | create_disks request', ['sakuracloud', 'volume']) do
  tests('success') do
    tests('#create_disks') do
      disks = sakuracloud_volume_service.create_disk('foobar', 4, {:sourcearchive => 112500463685})
      test 'returns a Hash' do
        disks.body.is_a? Hash
      end
```

File: lib/fog/sakuracloud/models/compute/servers.rb (hunk 3 above)
Comment (plribeiro3000) [inline lib/fog/sakuracloud/models/compute/servers.rb]: :+1:

File: lib/fog/sakuracloud/models/volume/disk.rb (hunk 2 above)
Comment (plribeiro3000) [inline lib/fog/sakuracloud/models/volume/disk.rb]: Ok. The this is something to be changed in another PR because it seems the whole codebase follow this pattern. :+1:

**Comment by noralife** [comment]
@plribeiro3000 Thank you for reviewing carefully.
My latest patch, a4822af, meets all your comments above.
Can you merge this patch?

File: lib/fog/sakuracloud/requests/volume/create_disk.rb (hunk 1 above)
Comment (noralife) [inline lib/fog/sakuracloud/requests/volume/create_disk.rb]: I committed as 85ebf05

File: lib/fog/sakuracloud/models/volume/disk.rb (hunk 2 above)
Comment (noralife) [inline lib/fog/sakuracloud/models/volume/disk.rb]: I committed as a4822af

**Comment by plribeiro3000** [comment]
@noralife I don't have merge permissions here. The code seems ok to me.

@sawanoboly Can you take a look and merge if you think its ok please?

Thanks!

**Comment by sawanoboly** [comment]
@noralife Thank you for your contribution :smile: 

@plribeiro3000 LGTM, Thanks!

**Comment by sawanoboly** [comment]
Released as v1.4.0.
